### ISSUE TITLE
Proto messages doesn't always produce canonical JSONs

### ISSUE THREAD

**Issue by doron-cohen** [bufbuild/protobuf-es#310]
The generated code for some types (e.g. Timestamp or enum) contain `toJson` and `toJsonString` methods. These seems to work when passing a Protobuf message object to `JSON.stringify`. But, when passing an object which contain such object as a property, this doesn't work as intended.

In our case `Timestamp` is serialized as an object of `seconds` and `nanos` and enums will serialize into their numbers.

We solved it by passing a `replacer` to `JSON.stringify` like:
```js
body: JSON.stringify(
    body, (_, value) => value?.toJson ? value.toJson() : value
),
```

According to [MDN's docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#description) `JSON.stringify` looks for the method `toJSON` and not `toJson` or `toJsonString`. Am I missing something?

**Comment by smaye81** [comment]
Hi @doron-cohen.  We actually just landed a change in https://github.com/bufbuild/protobuf-es/pull/306 which adds support for `JSON.stringify` by implementing a `toJSON` function.   It just hasn't been released yet.  Is this what you're looking for?

**Comment by doron-cohen** [comment]
Yes! Thanks for the quick reply. Is there a release planned soon?

**Comment by smaye81** [comment]
Yep.  I am planning to publish a release tomorrow (11/23).  Will post here when it's available.

**Comment by smaye81** [comment]
@doron-cohen v0.3.0 was released today, which includes the `toJSON` method on messages.  Full release notes [here](https://github.com/bufbuild/protobuf-es/releases/tag/v0.3.0).

Any issues, let us know.  Thanks!

**Comment by doron-cohen** [comment]
Already using it. Thanks @smaye81 !
